agent/pkg/singularity/singularity.go

```diff
@@ -310,9 +310,9 @@ func (s *SingularityClient) RunContainer(
 		}
 	}
 	if len(cudaVisibleDevices) > 0 {
-		// TODO(DET-9081): We need to move to --nvccli --nv, because --nv does not provide
-		// sufficient isolation (e.g., nvidia-smi see all GPUs on the machine, not just ours).
-		args = append(args, "--nv")
+		// Using --userns to avoid this error when using --nvccli:
+		// FATAL:   nvidia-container-cli not allowed in setuid mode
+		args = append(args, "--nv", "--nvccli", "--userns", "--contain")
```

Comment:
```
  -c, --contain                       use minimal /dev and empty other
                                      directories (e.g. /tmp and $HOME)
                                      instead of sharing filesystems from
                                      your host

```
Is this going to work for us?